components/fxa-client/src/oauth.rs

```diff
@@ -82,6 +82,54 @@ impl FirefoxAccount {
         Ok(token_info)
     }
 
+    /// Check whether access token is valid using the saved refresh token.
+    /// If there is no refresh token held or if it is not authorized,
+    /// Error is thrown.
+    pub fn check_authorization_status(&mut self, scope: &str) -> Result<IntrospectInfo> {
+        if scope.contains(' ') {
+            return Err(ErrorKind::MultipleScopesRequested.into());
+        }
+        let resp = match self.state.refresh_token {
+            Some(ref refresh_token) => {
+                if refresh_token.scopes.contains(scope) {
+                    self.client.oauth_introspection_with_refresh_token(
+                        &self.state.config,
+                        &refresh_token.token,
+                        &[scope],
+                    )?
+                } else {
+                    return Err(ErrorKind::NoCachedToken(scope.to_string()).into());
+                }
+            }
+            None => {
+                #[cfg(feature = "browserid")]
```

Comment:
You can probably get rid of all this branch now as this endpoint I believe will not work with session tokens.